### ISSUE TITLE
Clarify "Working with Files" regarding encoding in the docs

### ISSUE THREAD

**Issue by mark-summerfield** [JuliaLang/julia#20973]
In this part of the manual: http://docs.julialang.org/en/latest/manual/networking-and-streams.html#Working-with-Files-1 it doesn't mention file encoding at all. And Julia's file-related functions e.g., open() and eachline() don't seem to support an encoding argument. I think it would be helpful to say how Julia handles encodings -- e.g., does it default to UTF-8, and if so, how do you handle other encodings; and if not, then how do you end up with UTF-8-encoded Strings?

**Comment by nalimilan** [comment]
Yes, we should state very clearly that strings are expected to be in UTF-8. I'm not sure the manual often refers to packages, but it might be worth pointing at StringEncodings.jl and ICU.jl for when you need to handle other encodings.

> and if not, then how do you end up with UTF-8-encoded Strings?

What do you mean by that?

**Comment by StefanKarpinski** [comment]
For 1.0 we really need a clear story for how to deal with files and other I/O streams in different encodings. It should not entail having different string encodings in Base, but there should be a clear path to dealing with them that meshes well with what's in Base.
